broker/src/main/java/io/zeebe/broker/system/partitions/PartitionContext.java

```diff
@@ -277,4 +278,12 @@ public boolean pauseExporting() throws IOException {
   public boolean resumeExporting() throws IOException {
     return partitionProcessingState.resumeExporting();
   }
+
+  public void setPartition(final ZeebePartition zeebePartition) {
+    this.partition = zeebePartition;
+  }
+
+  public ZeebePartition getPartition() {
+    return partition;
+  }
```

Comment:
You can remove this. You can use the context.getComponentHealthMonitor() to register exporterDirector.